homeassistant/components/zha/config_flow.py

```diff
@@ -0,0 +1,54 @@
+"""Config flow for ZHA."""
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from .helpers import check_zigpy_connection
+from .const import (
+    DOMAIN, CONF_RADIO_TYPE, CONF_USB_PATH, RadioType
+)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZhaFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a zha config flow start."""
+        if self._async_current_entries():
+            return self.async_abort(reason='single_instance_allowed')
+
+        errors = {}
+
+        fields = OrderedDict()
+        fields[vol.Required(CONF_USB_PATH)] = str
+        fields[vol.Optional(CONF_RADIO_TYPE, default='ezsp')] = vol.In(
+            RadioType.list()
+        )
+
+        if user_input is not None:
+            test = await check_zigpy_connection(user_input[CONF_USB_PATH],
+                                                user_input[CONF_RADIO_TYPE])
+            if not test:
```

Comment:
Invert this check and return the create entry if true. Then we can remove the else and outdent the following statement.